lib/commands.coffee

```diff
@@ -51,12 +52,26 @@ class Commands
       atom.notifications.addError error.message, {detail: error.stack, dismissable: true}
 
   nextError: ->
-    if not @messages or (next = @messages.next()).done
-      next = (@messages = @linter.views.getMessages().values()).next()
-    return if next.done # There's no errors
-    message = next.value
-    return unless message.filePath
-    return unless message.range
+    if @index?
+      @index++
+    else
+      @index = 0
+    messages = @linter.views.messages
+    message = messages[@index %% messages.length]
```

Comment:
I find this kind of magic to be less than intuitive.  When you do something like this please enclose it in a method to explain what it does for those reading the code.  

Also just to make sure I interpreted this correctly.  You are basically taking the mod of the length so that just increasing the index value past the end will just loop around to the beginning. 

Maybe have a method called getCurrentMessageIndex()?